youtube_dl/extractor/youtube.py

```diff
@@ -1553,25 +1557,37 @@ def _extract_playlist(self, playlist_id):
                     'Invalid parameters. Maybe URL is incorrect.',
                     expected=True)
             elif re.match(r'[^<]*Choose your language[^<]*', match):
+                # Looks good; continue
                 continue
             else:
                 self.report_warning('Youtube gives an alert message: ' + match)
 
-        # Extract the video ids from the playlist pages
         def _entries():
+            # Extract the video ids from the playlist pages
             more_widget_html = content_html = page
+
             for page_num in itertools.count(1):
+                # Loop to find videos until break
+
                 matches = re.finditer(self._VIDEO_RE, content_html)
-                # We remove the duplicates and the link with index 0
-                # (it's not the first video of the playlist)
-                new_ids = orderedSet(m.group('id') for m in matches if m.group('index') != '0')
-                for vid_id in new_ids:
-                    yield self.url_result(vid_id, 'Youtube', video_id=vid_id)
 
+                # Make list of videos
+                new_videos = [{'id': m.group('id'),
+                               'title': m.group('title').strip()}
+                              for m in matches
+                              if m.group('index') != '0'  # Ignore link with index 0
+                              if m.group('title').strip()]  # Ignore links without titles, which also prevents duplicates
```

Comment:
Good points.  I fixed it to handle videos without titles.  It would be nice to use an OrderedDict to make the code for avoiding duplicate videos cleaner (because, when accepting videos without titles, it appears that every video that has a title ends up matching twice, once without and once with title; something weird about the HTML/JSON YouTube returns).

OrderedDicts are in Python 2.7.  Will we be able to use them in the future?